apps/dapp/src/components/Pages/Core/NewUI/OtcModal.tsx

```diff
@@ -0,0 +1,164 @@
+import { Popover } from 'components/Popover';
+import styled from 'styled-components';
+import { useEffect, useState } from 'react';
+import _ from 'lodash';
+import { Input } from './HomeInput';
+import { TICKER_SYMBOL } from 'enums/ticker-symbol';
+import { TradeButton } from './Home';
+import { useWallet } from 'providers/WalletProvider';
+import { formatToken } from 'utils/formatter';
+import { OtcOffer__factory, ERC20__factory } from 'types/typechain';
+import env from 'constants/env';
+import { getBigNumberFromString, getTokenInfo } from 'components/Vault/utils';
+import { useNotification } from 'providers/NotificationProvider';
+import { fromAtto } from 'utils/bigNumber';
+import { ethers } from 'ethers';
+
+interface IProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+const OHM = TICKER_SYMBOL.OHM;
+
+export const OtcModal: React.FC<IProps> = ({ isOpen, onClose }) => {
+  const { balance, wallet, updateBalance, signer, ensureAllowance } = useWallet();
+  const [input, setInput] = useState('');
+  const [output, setOutput] = useState('0');
+  const [allowance, setAllowance] = useState(0);
+  const { openNotification } = useNotification();
+
+  // Fetch the allowance for OtcOffer to spend OHM
+  const checkAllowance = async () => {
+    if (!signer || !wallet) return;
+    const ohmContract = new ERC20__factory(signer).attach(env.contracts.olympus);
+    const allowance = await ohmContract.allowance(wallet, env.contracts.otcOffer);
+    setAllowance(fromAtto(allowance));
```

Comment:
This shouldn't use `fromAtto` as that assumes 18dp. Should use something like `ethers.utils.formatUnits(allowance, getTokenInfo(OHM).decimals);`